From 840bb4b137a2e35cb30f4b78437b11b852ef6389 Mon Sep 17 00:00:00 2001
From: Jonas Bernoulli <jonas@bernoul.li>
Date: Sat, 19 Aug 2017 17:51:00 +0200
Subject: [PATCH] [PATCH] Add library wanderlust.el whose name matches that of
 the package wanderlust

A package should always contain a library with the same name.  One
benefit of that is that Melpa can then extract the package description
from library commentary.

Remove "wanderlust-pkg.el".  The only additional information it contains
is a version string, which has not ever been bumped since it was first
added in 2014.

[Emacsmirror] This commit is being rebased onto upstream.
---
 wanderlust-pkg.el |  6 ------
 wl/wanderlust.el  | 38 ++++++++++++++++++++++++++++++++++++++
 2 files changed, 38 insertions(+), 6 deletions(-)
 delete mode 100644 wanderlust-pkg.el
 create mode 100644 wl/wanderlust.el

diff --git a/wanderlust-pkg.el b/wanderlust-pkg.el
deleted file mode 100644
index 7d7361f39..000000000
--- a/wanderlust-pkg.el
+++ /dev/null
@@ -1,6 +0,0 @@
-(define-package "wanderlust" "2.15.9"
-  "Yet Another Message Interface on Emacsen"
-  '((emacs "24.5")
-    (apel "10.8")
-    (flim "1.14.9")
-    (semi "1.14.7")))
diff --git a/wl/wanderlust.el b/wl/wanderlust.el
new file mode 100644
index 000000000..055498505
--- /dev/null
+++ b/wl/wanderlust.el
@@ -0,0 +1,38 @@
+;;; wanderlust.el --- Yet Another Message Interface on Emacsen.
+
+;; Copyright (C) 1998-2000 Yuuichi Teranishi <teranisi@gohome.org>
+;; Copyright (C) 1998-2000 Masahiro MURATA <muse@ba2.so-net.ne.jp>
+
+;; Author: Yuuichi Teranishi <teranisi@gohome.org>
+;;	Masahiro MURATA <muse@ba2.so-net.ne.jp>
+;; Keywords: mail, net news
+;; Package-Requires: ((emacs "24.5") (apel "0") (flim "0") (semi "0"))
+
+;; This file is part of Wanderlust (Yet Another Message Interface on Emacsen).
+
+;; This program is free software; you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation; either version 2, or (at your option)
+;; any later version.
+;;
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+;;
+;; You should have received a copy of the GNU General Public License
+;; along with GNU Emacs; see the file COPYING.  If not, write to the
+;; Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+;; Boston, MA 02111-1307, USA.
+
+;;; Commentary:
+
+;; Wanderlust is a mail/news management system with IMAP4rev1 support
+;; for Emacs. It was originally developed by Yuuichi Teranishi.
+
+;;; Code:
+
+(require 'wl)
+(provide 'wanderlust)
+
+;;; wanderlust.el ends here
